### PR TITLE
Docker: add Dockerfile and docker-compose

### PR DESCRIPTION
**Pull request by jae1911**
## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Description

This pull-request adds Docker support for the application which allows for easier deployments.
You can now `docker build -t geoguess .` to build an image that you can easily use.
Everything is configurable with environment variables, just like on the baremetal install.

## How Has This Been Tested?

Tested by deploying it both locally and then on a production server using the provided example of `docker-compose`.
  
  
Signed off by: Jae Lo Presti <me@jae.fi>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM node:lts-alpine
+
+RUN npm install -g http-server
+WORKDIR /app
+COPY package*.json ./
+RUN apk add python3 make g++
```
Comment (BilelJegham) [inline Dockerfile:6]: Add those dependencies is really useful

**Review by BilelJegham — CHANGES_REQUESTED** [review]
Hello 👋,
Thanks you for your contribution.
Great idea to add docker support.

File: Dockerfile (hunk 1 above)
Comment (jae1911) [reply Dockerfile:6]: Yeah, it won't compile if those aren't there.
Testing if we can remove them after running the node install string.

Comment (jae1911) [reply Dockerfile]: Addressed this problem in https://github.com/GeoGuess/GeoGuess/pull/436/commits/a4c332272b6737924ae81b7e9278d9c7b70c42dc @BilelJegham

**Comment by jae1911** [comment]
The final Docker image is around 40MB which is an ideal size to deploy easily.
image

File: Dockerfile (hunk 1 above)
Comment (BilelJegham) [reply Dockerfile]: You right Pyhon & g++ need for node-gyp.

**Review by BilelJegham — APPROVED** [review]
Work great for me @jae1911

**Comment by BilelJegham** [comment]
@all-contributors please add @jae1911 for code

**Pull request merged**